cdap-app-fabric/src/main/java/io/cdap/cdap/internal/metadata/DefaultMetadataConsumerContext.java

```diff
@@ -16,32 +16,53 @@
 
 package io.cdap.cdap.internal.metadata;
 
+import io.cdap.cdap.api.metrics.MetricsCollectionService;
 import io.cdap.cdap.common.conf.CConfiguration;
 import io.cdap.cdap.common.conf.Constants;
 import io.cdap.cdap.spi.metadata.MetadataConsumerContext;
+import io.cdap.cdap.spi.metadata.MetadataConsumerMetrics;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
 
 /**
- * Provides an initialized default context for MetadataConsumer implementing {@link
- * MetadataConsumerContext}
+ * Provides an initialized default context for MetadataConsumer implementing {@link MetadataConsumerContext}.
  */
 public class DefaultMetadataConsumerContext implements MetadataConsumerContext {
 
   private final Map<String, String> properties;
+  private final String metadataConsumerName;
+  private final MetricsCollectionService metricsCollectionService;
+  private final String namespace;
 
   /**
+   * Constructs a new instance of ${@link DefaultMetadataConsumerContext}.
+   *
    * @param cConf An instance of an injected ${@link CConfiguration}.
    * @param metadataConsumerName name of the Metadata Consumer extension
+   * @param metricsCollectionService the underlying service for metrics publishing
+   * @param namespace namespace in which the program was run
    */
-  DefaultMetadataConsumerContext(CConfiguration cConf, String metadataConsumerName) {
-    String prefix = String.format("%s.%s.", Constants.MetadataConsumer.METADATA_CONSUMER_PREFIX,
-        metadataConsumerName);
+  DefaultMetadataConsumerContext(CConfiguration cConf, String metadataConsumerName,
+                                 MetricsCollectionService metricsCollectionService,
+                                 String namespace) {
+    String prefix = String.format("%s.%s.", Constants.MetadataConsumer.METADATA_CONSUMER_PREFIX, metadataConsumerName);
     this.properties = Collections.unmodifiableMap(cConf.getPropsWithPrefix(prefix));
+    this.metadataConsumerName = metadataConsumerName;
+    this.metricsCollectionService = metricsCollectionService;
+    this.namespace = namespace;
   }
 
   @Override
   public Map<String, String> getProperties() {
     return this.properties;
   }
+
+  @Override
+  public MetadataConsumerMetrics getMetrics(Map<String, String> context) {
+    Map<String, String> tags = new HashMap<>(context);
+    tags.put(Constants.Metrics.Tag.NAMESPACE, this.namespace);
+    tags.put(Constants.Metrics.Tag.METADATA_CONSUMER, metadataConsumerName);
```

Comment:
cehck if application name, program id and run id, etc are useful

Reply:
Useful to track success/failed calls for a pipeline, run and program.
Similar to - https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/DefaultProvisionerContext.java#L149